### ISSUE TITLE
Can't access /ipfs while /ipns works

### ISSUE THREAD

**Issue by rouanth** [ipfs/go-ipfs#3356]
#### Version information:
go-ipfs version: 0.4.4-
Repo version: 4
System version: amd64/linux
Golang version: go1.7.1

#### Type:
Bug

#### Priority:
P2

#### Description:
Can't access /ipfs after mount. /ipns works fine.

fusermount version: 2.9.7

I can reproduce the problem consistently by following these steps:

```sh
    rm -r ~/.ipfs
    sudo umount /ipns /ipfs # In case they were mounted previously
    sudo rm -r /ipns /ipfs
    sudo install -d -o `whoami` -m 755 /ipns /ipfs
    ipfs init
    ipfs daemon --mount &
    # Wait for IPFS to initialize
    ls -lh /ipns # lists the local@ symlink and my hash
    ls -lh /ipfs # ls: reading directory '/ipfs': Operation not permitted
```
My `/etc/fuse.conf` is empty, and `/dev/fuse` belongs to root:root. However, if it belongs to my_user:my_group, the error persists nevertheless.

My mountpoins after the specified commands are:

```
    fusectl on /sys/fs/fuse/connections type fusectl (rw,relatime)
    /dev/fuse on /ipfs type fuse (rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
    /dev/fuse on /ipns type fuse (rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
```

No errors are mentioned in the daemon logs.

Running the specified commands as root didn't solve the problem, so it's probably not a permissions-related problem.

I use Arch Linux and have go-ipfs 0.4.4-1 and fuse 2.9.7-1 packages installed.

**Comment by mateon1** [comment]
It isn't possible to `ls /ipfs` because that operation doesn't make sense, we can't print all IPFS hashes.
Accessing `/ipfs/QmHash...` should still work fine, though.

**Comment by rouanth** [comment]
Oh, I see. I've thought that I could use /ipfs/ to see the list of objects that are available locally, i.e. those that I've added, but I was wrong. For those who've made the same mistake as me — `ipfs pin ls` does the trick.

Closing the issue. But I believe that /ipfs/ would be a significantly more transparent concept if it worked like some versions of automounting software, e.g. autofs: when an object is first accessed, it's made available for listing. What's present now does the trick, however.

Thank you for clarification.
